hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeUsageInfo.java

```diff
@@ -60,25 +61,27 @@ private static int compareByUtilization(DatanodeUsageInfo first,
     if (first.equals(second)) {
       return 0;
     }
-    return Double.compare(first.calculateUtilization(),
-        second.calculateUtilization());
+    return Double.compare(first.calculateUtilization(0),
+        second.calculateUtilization(0));
   }
 
   /**
-   * Calculates utilization of a datanode. Utilization of a datanode is defined
-   * as its used space divided by its capacity. Here, we prefer calculating
+   * Calculates utilization of a datanode after adding a specified size.
+   * Utilization of a datanode is defined as its used space divided
+   * by its capacity. Here, we prefer calculating
    * used space as (capacity - remaining), instead of using
    * {@link SCMNodeStat#getScmUsed()} (see HDDS-5728).
    *
+   * @param plusSize the increased size
    * @return (capacity - remaining) / capacity of this datanode
    */
-  public double calculateUtilization() {
-    double capacity = scmNodeStat.getCapacity().get();
+  public double calculateUtilization(long plusSize) {
```

Comment:
NIT: Let's have another function `calculateUtilization()` which calls this function with 0 as parameter. Then we can keep using the older function in other places.

Reply:
sure, i will do this